dali/python/nvidia/dali/_multiproc/pool.py

```diff
@@ -202,10 +204,14 @@ def __init__(
             task_r, task_w = mp.Pipe(duplex=False)
             res_r, res_w = mp.Pipe(duplex=False)
             sock_reader, sock_writer = socket.socketpair()
+            if callback_pickler is None:
+                callbacks_ = callbacks
+            else:
+                callbacks_ = callback_pickler.dumps(callbacks)
```

Comment:
So now `callbacks` contains either pickled or raw callbacks?

Reply:
Yes, so in the first case it creates kind of additional layer of serialization where multiprocessing pickler just sees already serialized data. At first I simply set Dali customized pickler to be used directly by multiprocessing but it is unfortunately a global change for the whole process and: 
1. does not work if multiprocessing already started some actual work before DALI was able to modify the context
2. there's a question if that's not going to interfere with other packages using multiprocessing, like pytorch (it seemed to work okay, but I think that if we can avoid worrying about that let's avoid it.)
3. for user provided packages for serialization I still would go for this type of double serialization